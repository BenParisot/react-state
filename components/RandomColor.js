import React, { PureComponent } from 'react';

export default class RandomColor extends PureComponent {
  constructor() {
    super();
    this.state = {
      height: '100px',
      width: '100px',
      backgroundColor: 'black'
    };
  }

  componentDidMount() {
    this.timerID = setInterval(
      () => this.setColor(),
      1000
    );
  }

  componentWillUnmount() {
    clearInterval(this.timerID);
  }

  setColor() {
    this.setState({
      backgroundColor: generateRandomColor()
    });
  }

  render() {
    return (
      <div style={this.state}></div>
    );
  }

}

function generateRandomColor() {
  const chars = '0123456789ABCDEF';
  let color = '#';
  for(let i = 0; i < 6; i++) {
    color += chars[Math.floor(Math.random() * 16)];
  }
  return color;
}
