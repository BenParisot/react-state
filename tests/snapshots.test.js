import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from '../components/RandomColor';

describe('RandomColor component', () => {
  it('randomly generates a Color', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
