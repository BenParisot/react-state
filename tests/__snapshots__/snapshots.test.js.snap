// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomColor component randomly generates a Color 1`] = `
<div
  style={
    Object {
      "backgroundColor": "black",
      "height": "100px",
      "width": "100px",
    }
  }
/>
`;
